pkg/types/validation/installconfig.go

```diff
@@ -464,3 +465,33 @@ var (
 		return v
 	}()
 )
+
+func validateCloudCredentialsMode(mode types.CloudCredentialsMode, fldPath *field.Path, platform string) field.ErrorList {
+	if mode == "" {
+		return nil
+	}
+	validModes := map[types.CloudCredentialsMode]bool{}
+	switch platform {
+	case aws.Name, azure.Name, gcp.Name:
+		validModes[types.MintCloudCredentialsMode] = true
+		validModes[types.PassthroughCloudCredentialsMode] = true
+	}
+	switch platform {
+	case aws.Name:
+		validModes[types.ManualCloudCredentialsMode] = true
+	}
+	if validModes[mode] {
+		return nil
+	}
```

Comment:
hmm.. I would prefer this be static and a little more explicit ...

something like..
```
// allowedPlatformCredentialModes is a map of platform name and supported credential modes..
// for platform that does not exists in this map, should not be allowed to set any credential mode
allowedPlatformCredentialModes = map[string][]string{
aws.Name: []string{mint, passthrough},
gcp.Name: []string{mint, passthrough},
..
}
```